src/Utils/EndpointValidation.ts

```diff
@@ -0,0 +1,82 @@
+export function validateEndpoint(endpointToValidate: string, allowedEndpoints: ReadonlyArray<string>): boolean {
+    if (!endpointToValidate) {
+        return true;
+    }
+    const originToValidate: string = new URL(endpointToValidate).origin;
+    const allowedOrigins: string[] = allowedEndpoints.map(allowedEndpoint => new URL(allowedEndpoint).origin) || [];
+    return allowedOrigins.indexOf(originToValidate) >= 0;
+}
+
+export const allowedArmEndpoints: ReadonlyArray<string> = [
+    "https://​management.azure.com",
+    "https://​management.usgovcloudapi.net",
+    "https://management.chinacloudapi.cn"
+];
+
+export const allowedAadEndpoints: ReadonlyArray<string> = [
+    "https://login.microsoftonline.com/"
+];
+
+export const allowedParentFrameOrigins: ReadonlyArray<string> = [
+    `^https:\\/\\/cosmos\\.azure\\.(com|cn|us)$`,
+    `^https:\\/\\/[\\.\\w]*portal\\.azure\\.(com|cn|us)$`,
+    `^https:\\/\\/[\\.\\w]*portal\\.microsoftazure.de$`,
+    `^https:\\/\\/[\\.\\w]*ext\\.azure\\.(com|cn|us)$`,
+    `^https:\\/\\/[\\.\\w]*\\.ext\\.microsoftazure\\.de$`,
+    `^https://cosmos-db-dataexplorer-germanycentral.azurewebsites.de$`,
+];
+
+export const allowedJunoOrigins: ReadonlyArray<string> = [
+    "https://juno-test.documents-dev.windows-int.net",
+    "https://juno-test2.documents-dev.windows-int.net",
```

Comment:
Endpoint for juno test3 is missing. It should include `https://juno-test3.documents-dev.windows-int.net`

Reply:
There is a JunoEndpoints constants file from where we can use these endpoints too.